server/routes/api/user.js

```diff
@@ -1,21 +1,56 @@
 const express = require('express');
 const router = express.Router();
 
-/* GET users listing. */
-router.get('/', (req, res, next) => {
-  res.send(
-    'This URL will handle POST, PUT, and DELETE requests relating to user accounts.'
-  );
-});
+/* PATCH the user settings 
+req format:
+{
+  model-value: new-value
+}
+Locked values:
+* passwordHash
+*/
+router.patch('/', async (req, res, next) => {
+  let changed = false;
+  for (let key of Object.keys(req.body)) {
+    if (
+      typeof req.user.get(key) !== 'undefined' &&
+      key != 'passwordHash' &&
+      key != 'id' &&
+      key != 'role' &&
+      key != 'balance'
+    ) {
+      // TODO conver req.user to a db object
+      req.user.set(key, req.body[key]);
+      changed = true;
+    }
+  }
+
+  if (changed) {
+    req.user.save();
+    res.status(204);
```

Comment:
This just makes sure the request is flushed and transmitted to the client consistently. You should typically always call .send in route handlers at some point (unless you have a thrown error and want the default error handler to take care of it, or you can call next() and pass in the error to get to the error handling middlewares.)

```suggestion
    res.status(204).send();
```

Reply:
This is sent right outside the `if` statement, regardless of the status code.